packages/cli/src/commands/trigger.ts

```diff
@@ -0,0 +1,201 @@
+        ' F.ex. `--tags production,webapp --tags production,backend` will run checks with tags (production AND webapp) OR (production AND backend).',
+      multiple: true,
+      required: false,
+    }),
+    config: Flags.string({
+      char: 'c',
+      description: 'The Checkly CLI config filename.',
+    }),
+    timeout: Flags.integer({
+      default: 240,
+      description: 'A timeout (in seconds) to wait for checks to complete.',
+    }),
+    verbose: Flags.boolean({
+      char: 'v',
+      description: 'Always show the full logs of the checks.',
+      allowNo: true,
+    }),
+    reporter: Flags.string({
+      char: 'r',
+      description: 'A list of custom reporters for the test output.',
+      options: ['list', 'dot', 'ci', 'github'],
+    }),
+    env: Flags.string({
+      char: 'e',
+      description: 'Env vars to be passed to the check run.',
+      exclusive: ['env-file'],
+      multiple: true,
+      default: [],
+    }),
+    'env-file': Flags.string({
+      description: 'dotenv file path to be passed. For example --env-file="./.env"',
+      exclusive: ['env'],
+    }),
+    record: Flags.boolean({
+      description: 'Record check results in Checkly as a test session with full logs, traces and videos.',
+      default: false,
+    }),
+    'test-session-name': Flags.string({
+      char: 'n',
+      description: 'A name to use when storing results in Checkly with --record.',
+    }),
+  }
+
+  async run (): Promise<void> {
+    const { flags } = await this.parse(Trigger)
+    const {
+      location: runLocation,
+      'private-location': privateRunLocation,
+      config: configFilename,
+      tags: targetTags,
+      timeout,
+      verbose: verboseFlag,
+      record: shouldRecord,
+      reporter: reporterFlag,
+      env,
+      'env-file': envFile,
+      'test-session-name': testSessionName,
+    } = flags
+    const envVars = await getEnvs(envFile, env)
+    const { configDirectory, configFilenames } = splitConfigFilePath(configFilename)
+
+    let checklyConfig
+    try {
+      const { config } = await loadChecklyConfig(configDirectory, configFilenames)
+      checklyConfig = config
+    } catch (err) {} // Don't throw an error if the config file is missing
+    const location = await this.prepareRunLocation(checklyConfig?.cli, {
+      runLocation: runLocation as keyof Region,
+      privateRunLocation,
+    })
+    const verbose = this.prepareVerboseFlag(verboseFlag, checklyConfig?.cli?.verbose)
+    const reporterTypes = this.prepareReportersTypes(reporterFlag as ReporterType, checklyConfig?.cli?.reporters)
+    const reporters = createReporters(reporterTypes, location, verbose)
+
+    const repoInfo = getGitInformation()
+    const ciInfo = getCiInformation()
+
+    const runner = new TriggerRunner(
+      config.getAccountId(),
+      timeout,
+      verbose,
+      shouldRecord,
+      location,
+      targetTags?.map((tags: string) => tags.split(',')) ?? [],
```

Comment:
@clample , I know it's weird, but we allow users to include commas `,` inside tags. Is it possible to escape them if the user want to filter by a tag having a comma in it. If not, we could restrict the usage of `,` in tags.

Reply:
Good point @nahuelon 

I think that we can allow users to escape with `\,` if we come up with a regex.

Reply:
I played around with it, but didn't get a regex with escaping. Let's wait until someone asks for this.